sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -272,7 +272,7 @@ namespace Azure { namespace Core {
         return time_point() + static_cast<std::chrono::milliseconds>(msec);
       }
 
-      explicit ContextSharedState() : CancelAtMsecSinceEpoch(ToMsecSinceEpoch(time_point::max())) {}
+      explicit ContextSharedState() : CancelAtMsecSinceEpoch(ToMsecSinceEpoch((time_point::max)())) {}
```

Comment:
We should write and check-in a sample that we build in CI which includes `Windows.h` to ensure we stay green going forward.

https://github.com/Azure/azure-sdk-for-cpp/blob/50efbdb4c08ed57e7eedd1bd571dc377d3455c0a/samples/integration/cmake-fetch-content/src/main.cpp#L12

Reply:
Formatting + tests: https://github.com/CaseyCarter/azure-sdk-for-cpp/pull/1

Reply:
I've added a commit with a simple test change for each SDK that just adds a CPP that defines equivalent nasty macros before including the corresponding SDK's public header. Hopefully adding it to `azure-template` will be enough to ensure that future SDKs include coverage as well.